### ISSUE TITLE
FlushCommit does not return when an upstream commit is empty

### ISSUE THREAD

**Issue by derekchiang** [pachyderm/pachyderm#711]
This is because in PPS, we don't spawn jobs based on empty commits, so the corresponding downstream commits are never created, resulting in `FlushCommit` blocking forever.

The solution we are thinking is that we'd add a `RunEmpty` flag to the pipeline spec.  When the flag is set to true, jobs are run even if their inputs are empty.  If the flag is set to false, we don't run the job, but we still create the corresponding (empty) downstream commits.

So imagine a pipeline `A->B->C`.  If `A->B` outputs an empty commit in `B`, and `RunEmpty` is set to false, PPS will still create an empty commit in `C`.  That way `FlushCommit` will unblock.

**Comment by derekchiang** [comment]
Fixed

**Comment by JonathanFraser** [comment]
not completely, 1d5e2da13cf8164094eefec8c7df1caae361d940 "fixed" this by producing empty commits, however, the empty commits are un-parented even if they should be.

**Comment by derekchiang** [comment]
@JonathanFraser just merged #715 which fixed this new issue that you brought up.
